pkg/operator/ceph/nfs/config.go

```diff
@@ -126,3 +133,231 @@ RADOS_URLS {
 %url	` + url + `
 `
 }
+
+func ganeshaKrbConfigBlock(kerberosSpec *cephv1.KerberosSpec) string {
+	return fmt.Sprintf(`NFS_KRB5 {
+	PrincipalName = "%s" ;
+	KeytabPath = /etc/krb5.keytab ;
+	Active_krb5 = YES ;
+}
+`, kerberosSpec.GetPrincipalName())
+}
+
+func ganeshaConfigIncludeKrbBlock(nfs *cephv1.CephNFS, radosObjectName string) string {
+	// don't use sprintf b/c %u on front makes compiler confused
+	return `%url "rados://` + nfs.Spec.RADOS.Pool + `/` + nfs.Spec.RADOS.Namespace + `/` + radosObjectName + `"` + "\n\n"
+}
+
+func (r *ReconcileCephNFS) setRadosConfig(nfs *cephv1.CephNFS) error {
+	if nfs.Spec.Security.KerberosEnabled() {
+		return setKerberosRadosConfig(r.context, r.clusterInfo, nfs)
+	}
+	return removeKerberosRadosConfig(r.context, r.clusterInfo, nfs)
+}
+
+func setKerberosRadosConfig(context *clusterd.Context, clusterInfo *cephclient.ClusterInfo, nfs *cephv1.CephNFS) error {
+	radosPool := nfs.Spec.RADOS.Pool
+	radosNs := nfs.Spec.RADOS.Namespace
+	radosInfoStr := fmt.Sprintf("rados://%q/%q/", radosPool, radosNs)
+
+	// write ganesha kerberos configuration block into a temp file
+	krbBlockFile, err := ioutil.TempFile("/tmp", "krb-block-file")
+	if err != nil {
+		return errors.Wrapf(err, "failed to create temp file for ganesha kerberos configuration block for %s", radosInfoStr)
```

Comment:
`errors.Wrapf()` is old/deprecated, use `%w` for error substitution like

```go
fmt.Errorf("failed to create temp file for ganesha kerberos configuration block for %s: %w",
    radosInfoStr, err)
```

... but this might as well be some Rook convention that requires updating in many more places.

Reply:
It is a larger convention. It's funny that we once used the `message. %v` (similar) notation a long time ago and then broadly switched to the `errors.Wrapf()` to make it easier. @travisn may be interested in this note.

Reply:
@leseb went to a lot of work to convert to `errors.Wrap()`. The method isn't actually marked deprecated though, right? Anyway, we should think about that as a bigger PR instead of changing the pattern in this PR.